src/Common/src/Microsoft/Win32/SafeHandles/NtlmSecuritySafeHandles.cs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+using System.Security.Authentication.ExtendedProtection;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Microsoft.Win32.SafeHandles
+{
+    /// <summary>
+    /// Wrapper around a ntlm_buf*
+    /// </summary>
+    internal sealed class SafeNtlmBufferHandle : SafeHandle
+    {
+        public SafeNtlmBufferHandle()
+           : base(IntPtr.Zero, true)
+        {
+        }
+
+        public byte[] ToByteArray(int length, int offset)
+        {
+            Debug.Assert(length >= 0, "negative length of buffer");
+            Debug.Assert(offset < length, "invalid offset for  buffer");
```

Comment:
Is this assert correct?  Why does the offset need to be less than the length?  Maybe I'm in the wrong mindset, but typically when methods take an offset and a length, the method is starting at some offset and going for some length from that offset, e.g. Array.Copy(src, 0, dst, 5, 3) will copy to destination starting at offset 5 and will copy a length of 3 bytes.